client/my-sites/sharing/buttons/preview-button.jsx

```diff
@@ -30,10 +31,23 @@ module.exports = React.createClass( {
 	},
 
 	getIcon: function() {
-		if ( 'string' === typeof this.props.button.genericon ) {
-			return <span className="noticon sharing-buttons-preview-button__glyph" aria-hidden="true">{ formatting.unicodeToString( this.props.button.genericon ) }</span>;
+		const shortnameToSocialLogo = {
+			email: 'mail',
+			'google-plus-1': 'google-plus-alt',
+			pinterest: 'pinterest-alt',
+			tumblr: 'tumblr-alt',
+			'press-this': 'wordpress',
+			twitter: 'twitter-alt',
+			more: 'share'
+		}
+		if ( ! this.props.button.custom ) {
+			const icon = shortnameToSocialLogo[ this.props.button.ID ]
```

Comment:
You could shorten this line by using a simple `||` expression rather than a ternary:

```js
const icon = shortnameToSocialLogo[ this.props.button.ID ] || this.props.button.shortname;
```